packages/runtime/client/rest/src/index.ts

```diff
@@ -283,14 +301,30 @@ export function mkRestClient(baseURL: string): RestClient {
     (status) => status.version
   );
 
+  runtimeVersion.then((x) => {
+    if (CompatibleRuntimeVersionGuard.decode(x)._tag === "Left")
+      throw new Error(`Invalid runtime version ${x}`);
+  });
+
   return {
     healthcheck() {
       return healthcheck(axiosInstance);
     },
     version() {
-      return runtimeVersion;
+      return healthcheck(axiosInstance).then((status) => status.version);
     },
     getContracts(request) {
+      if (strict) {
+        const result = Contracts.GetContractsRequestGuard.decode(request);
+        if (result._tag === "Left") {
+          throw new InvalidTypeError(
+            result.left,
+            `Invalid argument to getContracts(${request})`
+          );
+        } else {
+          const test: typeof request = result.right;
```

Comment:
this seems unused

Reply:
It's unused at runtime but it helps ensure that you are using the correct guard. I think there are better ways of solving this. I'm going to look into it.